### DISCUSSION TITLE
Is `MatchesResult` useless code?

### DISCUSSION THREAD

**Discussion by BlackHole1** [containers/buildah#4897] [Q&A] [answered]
I would like to understand why this PR (https://github.com/containers/buildah/pull/1914) was not merged, while the closed PR depended on the upstream PR (https://github.com/containers/storage/pull/446). Moreover, the upstream PR (https://github.com/containers/storage/pull/446) has been merged, but the current situation resulted in the new function introduced in the PR (https://github.com/containers/storage/pull/446) not being utilized throughout the entire containers repository.

In https://github.com/containers/storage/pull/446, the Matches function has been added with a Deprecated comment. See: https://github.com/containers/storage/blob/8730e29911b3d1fc3122e8dca399782a138854e5/pkg/fileutils/fileutils.go#L60

/cc @saschagrunert @rhatdan @vrothberg

**Comment by vrothberg** [comment]
@flouthoc @nalind PTAL

**Comment by BlackHole1** [comment]
I'm currently in an uncomfortable situation because I noticed that the `Matches` function has been **deprecated**. As a result, all my changes were based on the `MatchesResult` function. After I finished modifying everything, I suddenly realized that none of the repositories under the "containers" organization were using this function except for an unmerged pull request. 🤣

**Comment by BlackHole1** [comment, marked as answer]
Close this discussion. See: https://github.com/containers/storage/pull/1656#discussion_r1247564933
